main/src/library/ReentrantLock.flix

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2021 Jonathan Lindegaard Starup
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+///
+/// A wrapper around a Java ReentrantLock.
+///
+pub opaque type ReentrantLock = ##java.util.concurrent.locks.ReentrantLock
+
+// TODO: waiting for Condition.flix to be merged into master
+opaque type Condition = ##java.util.concurrent.locks.Condition
+
+namespace ReentrantLock {
+
+    ///
+    /// Creates an instance of ReentrantLock with the given fairness policy.
+    /// `fair = False` is the default for Java.
+    ///
+    pub def makeLock(fair: Bool): ReentrantLock & Impure =
```

Comment:
In Flix, we tend to call these `new`, thus it becomes `ReetrantLock.new()` which locks nice, imo.